docformatter.py

```diff
@@ -380,6 +386,39 @@ def _find_shortest_indentation(lines):
 
     return indentation or ''
 
+def find_newline(source):
+    """Return type of newline used in source.
+    Input is a list of lines.
+    """
+    assert not isinstance(source, unicode)
+
+    counter = collections.defaultdict(int)
+    for line in source:
+        if line.endswith(CRLF):
+            counter[CRLF] += 1
+        elif line.endswith(CR):
+            counter[CR] += 1
```

Comment:
Add a unit test for this function that hits this line. Currently, it is not covered by any test.

Note that I already wrote exhaustive tests for this in `autopep8` that you could use.